lib/msf/core/post/windows/version.rb

```diff
@@ -0,0 +1,150 @@
+# -*- coding: binary -*-
+
+module Msf::Post::Windows::Version
+include Msf::Post::Windows::Registry
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Compat' => {
+          'Meterpreter' => {
+            'Commands' => %w[
+              stdapi_railgun_api
+            ]
+          }
+        }
+      )
+    )
+  end
+
+  def get_version_info
+    result = get_version_info_impl
+    if result.nil?
+      print_error("Couldn't retrieve the target's build number!")
+      raise RuntimeError.new("Couldn't retrieve the target's build number!")
+    end
+
+    result
+  end
+
+  def get_version_info_fallback_impl
+        build_num_raw = cmd_exec('ver')
+        groups = build_num_raw.match(/.*Version\s+(\d+)\.(\d+)\.(\d+)(\.(\d+))?/)
+        if groups.nil?
+          return nil
+        end
+
+        major, minor, build, unused, revision = groups.captures
```

Comment:
I just realized the capture group is there only because it is optional and it is never used. You can use `(?:…)` [syntax](https://ruby-doc.org/3.0.5/Regexp.html#class-Regexp-label-Grouping) to group without capturing:
```ruby
groups = build_num_raw.match(/.*Version\s+(\d+)\.(\d+)\.(\d+)(?:\.(\d+))?/)
major, minor, build, revision = groups.captures
```